src/test/java/fr/insee/pogues/transforms/visualize/PoguesJSONToPoguesJSONDerefImplTest.java

```diff
@@ -0,0 +1,82 @@
+package fr.insee.pogues.transforms.visualize;
+
+import fr.insee.pogues.model.ComponentType;
+import fr.insee.pogues.model.FlowControlType;
+import fr.insee.pogues.model.Questionnaire;
+import fr.insee.pogues.persistence.service.QuestionnairesService;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class PoguesJSONToPoguesJSONDerefImplTest {
+
+    /**
+     * Tested questionnaire: 'lct78jr8'
+     * This questionnaire contains references to the following questionnaires:
+     * - 'l4i3m6qa' (with a filter condition in the root questionnaire)
+     * - 'l6dnlrka' (simple questionnaire)
+     * - 'lct8pcsy' (including loops) */
+    @Test
+    void testJsonQuestionnaireComposition() throws Exception {
```

Comment:
Could it not be better to have 3 tests here ? one for simple questionnaire, one for loops and one for filter ?